domain.go

```diff
@@ -113,8 +113,34 @@ type DomainInterface struct {
 	Driver *DomainInterfaceDriver `xml:"driver"`
 }
 
+type DomainChardevSource struct {
+	Mode string `xml:"mode,attr"`
+	Path string `xml:"path,attr"`
+}
+
+type DomainChardevTarget struct {
+	Type  string `xml:"type,attr"`
+	Name  string `xml:"name,attr"`
+	State string `xml:"state,attr,omitempty"` // is guest agent connected?
+}
+
+type DomainAlias struct {
+	Name string `xml:"name,attr"`
+}
+
+type DomainAddress struct {
+	Type       string `xml:"type,attr"`
+	Controller string `xml:"controller,attr"`
+	Bus        string `xml:"bus,attr",omitempty`
+	Port       string `xml:"port,attr,omitempty"`
```

Comment:
Based on our discussions in other PR, we should go with '* uint' for these types, rather than strings, so we have strongly typed structs.